modules/hl7802/cellular_hl7802_api.c

```diff
@@ -541,27 +625,59 @@ static CellularError_t buildSocketConfig( CellularSocketHandle_t socketHandle,
-
         /* Set the local port in the end of command buffer string if localPort is not 0. */
         if( socketHandle->localPort > 0 )
         {
             ( void ) snprintf( portBuf, sizeof( portBuf ),
                                ",%u",
                                socketHandle->localPort );
+        }
 
-            /* Because the length of host's IP address is limited,
-             * the buffer size must be enough for port setting. */
-            strcat( pCmdBuf, portBuf );
+        /* Format: AT+KTCPCFG=[<cnx cnf>],<mode>,[<tcp remote address>],<tcp_port>[,[<source_port>]
+         * [,[<data_mode>][,[<URC-ENDTCP-enable>][,[<af>][,[<cipher_suite>][,[<restore_on_boot>]]]]]]]] */
+        /* To set TCP connection, remote address & port should be put before local port. */
+        if( socketHandle->socketProtocol == CELLULAR_SOCKET_PROTOCOL_TCP )
+        {
+            ( void ) snprintf( remoteInfoBuf, sizeof( remoteInfoBuf ),
+                               ",\"%s\",%u",
+                               socketHandle->remoteSocketAddress.ipAddress.ipAddress,
+                               socketHandle->remoteSocketAddress.port );
+
+            /* The return value of snprintf is not used.
+             * The max length of the string is fixed and checked offline.
+             * Reserve buffer for port setting. */
+            /* coverity[misra_c_2012_rule_21_6_violation]. */
+            ( void ) snprintf( pCmdBuf, CELLULAR_AT_CMD_MAX_SIZE,
+                               "%s=%u,0%s%s",
+                               "AT+KTCPCFG",
+                               socketHandle->contextId,
+                               remoteInfoBuf,
+                               portBuf );
+        }
+
+        /* Format: AT+KUDPCFG=[<cnx cnf>],<mode>[,[<port>][,[<data_mode>][,[<udp remote address>]
+         * [,[<udp_port>][,[<af>][,[<restore_on_boot>]]]]]]] */
+        /* To set UDP connection, remote address & port should be put after local port. */
+        else
+        {
+            if( strlen( socketHandle->remoteSocketAddress.ipAddress.ipAddress ) > 0 )
+            {
+                ( void ) snprintf( remoteInfoBuf, sizeof( remoteInfoBuf ),
+                                   ",\"%s\",%u",
+                                   socketHandle->remoteSocketAddress.ipAddress.ipAddress,
+                                   socketHandle->remoteSocketAddress.port );
+            }
+
+            /* The return value of snprintf is not used.
+             * The max length of the string is fixed and checked offline.
+             * Reserve buffer for port setting. */
+            /* coverity[misra_c_2012_rule_21_6_violation]. */
+            ( void ) snprintf( pCmdBuf, CELLULAR_AT_CMD_MAX_SIZE,
+                               "%s=%u,0%s%s",
+                               "AT+KUDPCFG",
+                               socketHandle->contextId,
+                               portBuf,
+                               remoteInfoBuf );
```

Comment:
I can't do this because the local port is put at different sequence.
In TCP -> AT+KTCPCFG=[cnx cnf],mode,**[tcp remote address],tcp_port[,[source_port]...**
In UDP -> AT+KUDPCFG=[cnx cnf],mode[,[port][,[data_mode][,[udp remote address][,[udp_port]...**